### PR TITLE
refactor(updatenotification): Migrate legacy code

### PR DESCRIPTION
**Pull request by susnux**
## Summary

1. Remove hook usage and just provide an initial state
2. Replace jQuery code with modern non-deprecated frontend code

Also fixed the appinfo.xml as the order of entries is fixed in the XSD.

### Screen shots
Bildschirmfoto_20240924_170829

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by provokateurin — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

File: apps/updatenotification/lib/UpdateChecker.php [hunk 1]
```diff
@@ -59,13 +55,17 @@
...
 	/**
-	 * @param array $data
+	 * Provide update information as initial state
 	 */
-	public function populateJavaScriptVariables(array $data) {
-		$data['array']['oc_updateState'] = json_encode([
-			'updateAvailable' => true,
-			'updateVersion' => $this->getUpdateState()['updateVersionString'],
-			'updateLink' => $this->getUpdateState()['updateLink'] ?? '',
+	public function setInitialState() {
```
[the remark this answers is not in the document]
Comment (susnux) [reply apps/updatenotification/lib/UpdateChecker.php:60]:
```suggestion
	public function setInitialState(): void {
```

File: apps/updatenotification/src/update-notification-legacy.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/**
```
Comment (come-nc) [inline apps/updatenotification/src/update-notification-legacy.ts:1]: Why is the file still named with -legacy?

Comment (susnux) [reply apps/updatenotification/src/update-notification-legacy.ts:1]: Legacy did not meant the content but the usage: It is only needed if you do not have notifications app enabled.

We could also name it `-fallback`

Comment (come-nc) [reply apps/updatenotification/src/update-notification-legacy.ts:1]: fallback would be clearer for me, but no big deal, was just my curiosity when reading this.

**Pull request merged**